packages/extension-koni-ui/src/Popup/Home/Staking/StakeCompoundSubmitTransaction.tsx

```diff
@@ -0,0 +1,500 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import { DelegationItem } from '@subwallet/extension-base/background/KoniTypes';
+import cloneLogo from '@subwallet/extension-koni-ui/assets/clone.svg';
+import Button from '@subwallet/extension-koni-ui/components/Button';
+import InputAddress from '@subwallet/extension-koni-ui/components/InputAddress';
+import InputBalance from '@subwallet/extension-koni-ui/components/InputBalance';
+import Spinner from '@subwallet/extension-koni-ui/components/Spinner';
+import { ActionContext } from '@subwallet/extension-koni-ui/contexts';
+import useGetNetworkJson from '@subwallet/extension-koni-ui/hooks/screen/home/useGetNetworkJson';
+import useIsNetworkActive from '@subwallet/extension-koni-ui/hooks/screen/home/useIsNetworkActive';
+import useToast from '@subwallet/extension-koni-ui/hooks/useToast';
+import useTranslation from '@subwallet/extension-koni-ui/hooks/useTranslation';
+import { checkTuringStakeCompounding, getStakeDelegationInfo, getTuringCancelStakeCompoundTxInfo, getTuringStakeCompoundTxInfo } from '@subwallet/extension-koni-ui/messaging';
+import Header from '@subwallet/extension-koni-ui/partials/Header';
+import { RootState } from '@subwallet/extension-koni-ui/stores';
+import { ThemeProps } from '@subwallet/extension-koni-ui/types';
+import { toShort } from '@subwallet/extension-koni-ui/util';
+import { formatLocaleNumber } from '@subwallet/extension-koni-ui/util/formatNumber';
+import moment from 'moment';
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import styled from 'styled-components';
+
+import { BN } from '@polkadot/util';
+
+const StakeAuthCompoundRequest = React.lazy(() => import('@subwallet/extension-koni-ui/Popup/Home/Staking/components/StakeAuthCompoundRequest'));
+const StakeCompoundResult = React.lazy(() => import('@subwallet/extension-koni-ui/Popup/Home/Staking/components/StakeCompoundResult'));
+const StakeAuthCancelCompoundRequest = React.lazy(() => import('@subwallet/extension-koni-ui/Popup/Home/Staking/components/StakeAuthCancelCompoundRequest'));
+const ValidatorsDropdown = React.lazy(() => import('@subwallet/extension-koni-ui/Popup/Bonding/components/ValidatorsDropdown'));
+const CancelStakeCompoundResult = React.lazy(() => import('@subwallet/extension-koni-ui/Popup/Home/Staking/components/CancelStakeCompoundResult'));
+
+interface Props extends ThemeProps {
+  className?: string;
+}
+
+function filterValidDelegations (delegations: DelegationItem[]) {
+  const filteredDelegations: DelegationItem[] = [];
+
+  delegations.forEach((item) => {
+    if (parseFloat(item.amount) > 0) {
+      filteredDelegations.push(item);
+    }
+  });
+
+  return filteredDelegations;
+}
+
+const TURING_ED = 0.01; // TODO: can be done better, fix this upon new architecture
+
+function StakeCompoundSubmitTransaction ({ className }: Props): React.ReactElement<Props> {
```

Comment:
Please use useReducer instead of multiple useState for this component

Reply:
Will be updated in the future along with other components